WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -231,15 +274,100 @@ public void hideLoadingProgress() {
         }
     }
 
+    /*
+     * use this to add custom control/component views to your toolbar. Components are added on the right side of the
+      * toolbar */
+    public void addToolbarCustomControl(View v, OnClickListener clickListener){
+        if (v != null){
+            mCustomComponentsContainer.addView(v);
+            if (clickListener != null){
+                v.setOnClickListener(clickListener);
+            }
+        }
+    }
+
+    public void setToolbarBackgroundColor(int color){
+        mToolbar.setBackgroundColor(color);
+    }
+
+    public void setToolbarSpinnerTextColor(int color){
+        mSpinnerTextColor = color;
+    }
+
+    public void setToolbarSpinnerDrawable(int drawableResId){
+        mSpinnerDrawableRight = drawableResId;
+    }
+
+    public void setToolbarLeftPadding(int paddingLeft){
+        mToolbar.setPadding(paddingLeft,
+                mToolbar.getPaddingTop(),
+                mToolbar.getPaddingRight(),
+                mToolbar.getPaddingBottom());
+    }
+
+    public void setToolbarRightPadding(int paddingRight){
+        mToolbar.setPadding(
+                mToolbar.getPaddingLeft(),
+                mToolbar.getPaddingTop(),
+                paddingRight,
+                mToolbar.getPaddingBottom());
+    }
+
+    public void setToolbarLeftAndRightPadding(int paddingLeft, int paddingRight){
+        mToolbar.setPadding(
+                paddingLeft,
+                mToolbar.getPaddingTop(),
+                paddingRight,
+                mToolbar.getPaddingBottom());
+    }
+
+    public void scrollRecycleViewToPosition(int position) {
+        if (mRecyclerView == null) return;
+
+        mRecyclerView.scrollToPosition(position);
+    }
+
+    public int getCurrentPosition() {
+        if (mRecyclerView != null && mRecyclerView.getLayoutManager() != null) {
+            return ((LinearLayoutManager) mRecyclerView.getLayoutManager()).findFirstVisibleItemPosition();
+        } else {
+            return -1;
+        }
+    }
+
+    public void smoothScrollToPosition(int position){
+        if (mRecyclerView != null && mRecyclerView.getLayoutManager() != null) {
+            mRecyclerView.getLayoutManager().smoothScrollToPosition(mRecyclerView, null, position);
+        }
+    }
+
+    /*
+    * use this to access the internal RecyclerView if you need to, for example, add items decoration
+    * */
+    public RecyclerView getInternalRecyclerView(){
```

Comment:
Definitely a tough call here! Yes it _is_ ugly :)  but I'll cite my initial PR comment as I could see this was going to cause some noise:

> getInternalRecyclerView allows to get the internal RecyclerView instance to manipulate it. You need access to this for example when item decoration need be added (and it didn’t make sense to re-expose the same API with the need to maintain it in the future)

While it also seemed ugly to me at first, it didn't make sense to make a wrapper method that adds no value and has to be maintained in the future, just for adding item decorations. Also the internal RecyclerView could be needed for something else other than just item decorations (setting a scroll changed listener maybe? ) and in the long run we would end up adding methods to wrap the internal methods only. Didn't feel right either.

Will add it if you feel it's best - just wanted to clear up what my thinking process was about this